### PR TITLE
add vearch to vector store

### PR DESCRIPTION
**Pull request by zhanghexian**
- [ ] New feature (non-breaking change which adds functionality)

- [ ] Added new notebook (that tests end-to-end)
- [ ] I stared at the code and made sure it makes sense

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

**Comment by logan-markewich** [comment]
@zhanghexian can you follow the new contribution guide? Specifically, you should create a package in `llama-index-integrations/vector_stores` instead of modifying legacy
https://docs.llamaindex.ai/en/stable/contributing/contributing.html#creating-a-new-integration-package

**Comment by zhanghexian** [comment]
Sorry, I don't quite understand how to package based on my submitted code, just execute this command:
cd ./llama-index-integrations/llms
llamaindex-cli new-package --kind "llms" --name "vearch"?
I'm very sorry to trouble you!

**Comment by zhanghexian** [comment]
I'm very sorry to trouble you. Could you be more specific? I still don't know how to operate the document when I read it.

**Comment by logan-markewich** [comment]
@zhanghexian you can take a look at other vector store integrations for an idea on structure 

to create a boostrapped folder 

```
cd llama-index-integrations/vector_stores
llamaindex-cli new-package --kind "vector_stores" --name "vearch"
```

We want to add the integration there, and not modify anything in llama-index-legacy

**Comment by zhanghexian** [comment]
I am very sorry to trouble you, may I ask which version of llama-index can run this command(llamaindex-cli new-package --kind "vector_stores" --name "vearch")? The latest version of llama-index requires onnxruntime>1.17.0, but there is no package of this version on the company Intranet at present. I tried a lower version of llama-index, but couldn't run this command: llamaindex-cli

**Comment by ravi03071991** [comment]
@zhanghexian which python version are you on? It will work on python 3.11 version.

**Comment by zhanghexian** [comment]
image
Now I have the python environment set to 3.11, but I still have the same problem, I tried to change the download  source and install the .whl file directly, but failed!

**Comment by zhanghexian** [comment]
The problem is that when llama-index=0.10.15 is installing, the llama-index-cli version is incompatible with the onnxruntime version, and the onnxruntimet needs to be installed.>=1,17.0 Please tell me which version of llama-index is suitable to install,thanks!

**Comment by zhanghexian** [comment]
now i commit it use llamaindex-cli,please check thanks！

File: llama-index-integrations/vector_stores/llama-index-vector-stores-vearch/llama_index/vector_stores/vearch/base.py [hunk 1]
```diff
@@ -0,0 +1,467 @@
...
+        """
+        ret: Optional[bool] = None
+        tmp_res = []
+        if ids is None or len(ids) == 0:
+            return ret
+        for _id in ids:
+            if self.flag:
+                ret = self.vearch.delete(self.using_db_name, self.using_table_name, _id)
+            else:
+                ret = self.vearch.del_doc(_id)
```
Comment (logan-markewich) [inline llama-index-integrations/vector_stores/llama-index-vector-stores-vearch/llama_index/vector_stores/vearch/base.py:450]: Actually, the intention behind delete is to delete any node that has `ref_doc_id`/`_id` in its metadata under `doc_id` -- is this possible in vearch?

Comment (zhanghexian) [reply llama-index-integrations/vector_stores/llama-index-vector-stores-vearch/llama_index/vector_stores/vearch/base.py:450]: vearch support deletion Specified id not all,thanks!

Comment (logan-markewich) [reply llama-index-integrations/vector_stores/llama-index-vector-stores-vearch/llama_index/vector_stores/vearch/base.py:450]: So then this function isn't working properly -- it should be deleting every node that has `ref_doc_id` in its metadata under `doc_id` 👀

Comment (zhanghexian) [reply llama-index-integrations/vector_stores/llama-index-vector-stores-vearch/llama_index/vector_stores/vearch/base.py:450]: Can you tell me what ref_doc_id is and the difference between doc_id? There may be some ambiguity in my understanding. vearch will delete all information corresponding to Doc_id according to DOC_ID

Comment (zhanghexian) [reply llama-index-integrations/vector_stores/llama-index-vector-stores-vearch/llama_index/vector_stores/vearch/base.py:450]: _id in this code means doc_id which is user want to delete !

Comment (logan-markewich) [reply llama-index-integrations/vector_stores/llama-index-vector-stores-vearch/llama_index/vector_stores/vearch/base.py:450]: So. When a document is broken into nodes, each node contains a reference to its parent doc `node.ref_doc_id`

When we call delete in a vector store, we want to delete all nodes that belong to a ref_doc_id

It's stored in the node metadata under a few names during the node_to_metadata_dict function, `doc_id` is one of those names I picked.

In any case, the idea is, we want to do a filtered delete here

**Comment by zhanghexian** [comment]
thanks,fixed! please check!

**Review by logan-markewich — APPROVED** [review]

**Comment by zhanghexian** [comment]
hello,i saw the sigh that said some conficts in .md file ,Do I need to make any changes? I'm sorry to trouble you, and when can I see the code I submitted in the project.thanks!

**Pull request merged**